sdk/cosmos/azure-cosmos/test/test_crud_async.py

```diff
@@ -0,0 +1,2647 @@
+
+#  	To Run the test, replace the two member fields (masterKey and host) with values
+#   associated with your Azure Cosmos account.
+
+
+class TimeoutTransport(RequestsTransport):
+
+    def __init__(self, response):
+        self._response = response
+        super(TimeoutTransport, self).__init__()
+
+    async def send(self, *args, **kwargs):
+        if kwargs.pop("passthrough", False):
+            return super(TimeoutTransport, self).send(*args, **kwargs)
+
+        time.sleep(5)
+        if isinstance(self._response, Exception):
+            raise self._response
+        output = requests.Response()
+        output.status_code = self._response
+        response = RequestsTransportResponse(None, output)
+        return response
+
+
+@pytest.mark.usefixtures("teardown")
+class CRUDTests(unittest.TestCase):
+    """Python CRUD Tests.
+    """
+
+    last_headers = []
+
+    async def __AssertHTTPFailureWithStatus(self, status_code, func, *args, **kwargs):
+        """Assert HTTP failure with status.
+
+        :Parameters:
+            - `status_code`: int
+            - `func`: function
+        """
+        try:
+            await func(*args, **kwargs)
+            self.assertFalse(True, 'function should fail.')
+        except exceptions.CosmosHttpResponseError as inst:
+            self.assertEqual(inst.status_code, status_code)
+
+    @classmethod
+    async def setUpClass(cls):
+        cls.client = cosmos_client.CosmosClient(cls.host, cls.masterKey, consistency_level="Session", connection_policy=cls.connectionPolicy)
+        cls.databaseForTest = await cls.client.create_database_if_not_exists(test_config._test_config.TEST_DATABASE_ID)
+
+    async def test_database_crud(self):
+        # read databases.
+        databases = [database async for database in self.client.list_databases()]
+        # create a database.
+        before_create_databases_count = len(databases)
+        database_id = str(uuid.uuid4())
+        created_db = await self.client.create_database(database_id)
+        self.assertEqual(created_db.id, database_id)
+        # Read databases after creation.
+        databases = [database async for database in self.client.list_databases()]
+        self.assertEqual(len(databases),
+                         before_create_databases_count + 1,
+                         'create should increase the number of databases')
+        # query databases.
+        databases = [database async for database in self.client.query_databases(
+            query='SELECT * FROM root r WHERE r.id=@id',
+            parameters=[
+                {'name': '@id', 'value': database_id}
+            ]
+        )]
+
+        self.assertTrue(len(databases) > 0)
+
+        # read database.
+        self.client.get_database_client(created_db.id)
+
+        # delete database.
+        await self.client.delete_database(created_db.id)
+        # read database after deletion
+        read_db = self.client.get_database_client(created_db.id)
```

Comment:
Q: I thought get_database_client will only get a databaseProxy, will it also read the database?

Reply:
gotcha, thanks